can/interfaces/nican.py

```diff
@@ -0,0 +1,282 @@
+    nican = ctypes.windll.LoadLibrary("nican")
+except Exception as e:
+    logger.error("Failed to load NI-CAN driver: %s", e)
+else:
+    nican.ncConfig.argtypes = [
+        ctypes.c_char_p, ctypes.c_ulong, ctypes.c_void_p, ctypes.c_void_p]
+    nican.ncConfig.errcheck = check_status
+    nican.ncOpenObject.argtypes = [ctypes.c_char_p, ctypes.c_void_p]
+    nican.ncOpenObject.errcheck = check_status
+    nican.ncCloseObject.errcheck = check_status
+    nican.ncAction.argtypes = [ctypes.c_ulong, ctypes.c_ulong, ctypes.c_ulong]
+    nican.ncAction.errcheck = check_status
+    nican.ncRead.errcheck = check_status
+    nican.ncWrite.errcheck = check_status
+    nican.ncWaitForState.argtypes = [
+        ctypes.c_ulong, ctypes.c_ulong, ctypes.c_ulong, ctypes.c_void_p]
+    nican.ncWaitForState.errcheck = check_status
+    nican.ncStatusToString.argtypes = [
+        ctypes.c_int, ctypes.c_uint, ctypes.c_char_p]
+
+
+class NicanBus(BusABC):
+    """
+    The CAN Bus implemented for the NI-CAN interface.
+    """
+
+    def __init__(self, channel, **kwargs):
+        """
+        :param str channel:
+            Name of the object to open (e.g. 'CAN0')
+
+        :param int bitrate:
+            Bitrate in bits/s
+
+        :param list can_filters:
+            A list of dictionaries each containing a "can_id" and a "can_mask".
+            Must only contain one filter!
+
+            >>> [{"can_id": 0x11, "can_mask": 0x21}]
+
+        :param int read_queue:
+            Length of read queue
+
+        :param int write_queue:
+            Length of write queue
+
+        :param bool log_errors:
+            If True, communication errors will appear as CAN messages with
+            ``is_error_frame`` set to True and ``arbitration_id`` will identify
+            the error (default True)
+
+        :raises can.interfaces.nican.NicanError:
+            If starting communication fails
+        """
+        self.channel_info = "NI-CAN: " + channel
+        if not isinstance(channel, bytes):
+            channel = channel.encode()
+
+        config = {
+            NC_ATTR_START_ON_OPEN: True,
+            NC_ATTR_CAN_COMP_STD: 0,
+            NC_ATTR_CAN_MASK_STD: 0,
+            NC_ATTR_CAN_COMP_XTD: 0,
+            NC_ATTR_CAN_MASK_XTD: 0,
+        }
+        can_filters = kwargs.get("can_filters")
+        if not can_filters:
+            logger.info("Filtering has been disabled")
+        elif len(can_filters) == 1:
+            can_id = can_filters[0]["can_id"]
+            can_mask = can_filters[0]["can_mask"]
+            logger.info("Filtering on ID 0x%X, mask 0x%X", can_id, can_mask)
+            # This did not filter out anything when tested but I don't know why
+            config[NC_ATTR_CAN_COMP_STD] = can_id
+            config[NC_ATTR_CAN_MASK_STD] = can_mask
+            config[NC_ATTR_CAN_COMP_XTD] = can_id
+            config[NC_ATTR_CAN_MASK_XTD] = can_mask
+        else:
+            logger.warning("Only one filter is supported")
```

Comment:
Do you mean a logging error or an exception? If the filter is used as an optimization then it will probably not be a big deal, just that the application uses unnecessary resources processing all messages on the bus.